tee-worker/omni-executor/aa-contracts/src/accounts/OmniAccount.sol

```diff
@@ -128,10 +145,32 @@ contract OmniAccount is BaseAccount, TokenCallbackHandler, UUPSUpgradeable, Init
         if (signer == UserOpSigner.Owner) {
             return _validateOwner(userOpHash, sig);
         } else if (signer == UserOpSigner.RootKey) {
+            // For non-EVM owner types, allow root signers to call restricted functions only if no passkey signers exist
+            if (ownerType != OwnerType.Evm && passkeySignerCount == 0) {
+                // Root signers can act as owners for non-EVM accounts without passkey signers
+                return _validateRootKey(userOpHash, sig);
+            }
+            // Otherwise, check if the UserOp is trying to call a restricted function
+            if (_isRestrictedCall(userOp.callData)) {
+                return SIG_VALIDATION_FAILED;
+            }
             return _validateRootKey(userOpHash, sig);
         } else if (signer == UserOpSigner.SessionKey) {
+            // Session keys should also be restricted from sensitive operations
+            if (_isRestrictedCall(userOp.callData)) {
+                return SIG_VALIDATION_FAILED;
+            }
             return _validateSessionKey(userOpHash, sig);
         } else if (signer == UserOpSigner.Passkey) {
+            // For non-EVM owner types with passkey signers, allow passkeys to call restricted functions
+            if (ownerType != OwnerType.Evm && passkeySignerCount > 0) {
+                // Passkey signers can act as owners for non-EVM accounts
+                return _validatePasskey(userOpHash, sig);
+            }
+            // Otherwise, check if the UserOp is trying to call a restricted function
+            if (_isRestrictedCall(userOp.callData)) {
+                return SIG_VALIDATION_FAILED;
+            }
```

Comment:
There's a edge case that can be a design debate: an EVM-owner could also bind a passkey: confirming with FaceID can be much more convenient than wallet signing.

If we treat passkey as privileged as owner - here we should simplify the logic to just `return _validatePasskey(userOpHash, sig);`

Reply:
True - just for consistency